core/domain/exp_domain.py

```
@@ -4224,6 +4222,9 @@ def validate(self):
             raise Exception(
                 'Assigned state ids should be smaller than last state id used.')
 
+        if not all(isinstance(x, int) for x in state_ids):
+            raise Exception('Assigned state ids should be integer values.')
```

Why wouldn't this be the case?

---

I have no idea but last time when I checked production models I encountered `null` values being set for END states.

---

That may indicate invalid explorations; we should probably output the IDs of the explorations that fail this check so that we can investigate them. Silencing this error may cause broader problems downstream; it's not clear if this migration can be safely done if this invariant doesn't hold.

---

That is a good idea. I will modify this in next commit.

---

Done